pkg/util/rolloutblock/blocks.go

```diff
@@ -63,18 +61,34 @@ func BlocksRollout(rolloutBlockLister shipperlisters.RolloutBlockLister, obj met
 	}
 }
 
-func ValidateOverrides(overrides ObjectNameList) error {
-	if len(overrides) == 0 {
+func ProcessBlocks(existingBlocks ObjectNameList, overrides ObjectNameList) error {
```

Comment:
I suggest to use more consistent naming for arguments of the same time: e.g.:
```go
func ProcessBlocks(existing, overrides ObjectNameList)
```

The method name needs a comment: the normal workflow here is to return an error, and the meaning of the nil return result is not trivial. I believe we might leave a comment explaining what return values mean.

The method name doesn't really process blocks: it figures out if it's good to go or not. From this perspective we might consider renaming it to something more self-explanatory.